### PR TITLE
Publish playback state paired with its media generation (#89)

### PR DESCRIPTION
**Pull request by omaralbeik**
Addresses **#89 criterion 2** — "a late subscriber immediately receives current media generation and state".

## Problem

`stateTransitions` cannot answer two questions its consumers keep asking.

A subscriber that arrives late learns nothing until the next change, which for a settled or idle player may be never. And a repeated `.playing` after same-player media replacement is indistinguishable from the first. The existing doc comment already names this, deferring it *"until events carry a media generation to distinguish sessions by"*.

## API

```swift
public var generation: PlaybackGeneration                       // current session
public nonisolated var playbackStatus: AsyncStream<PlaybackStatus>
```

`PlaybackStatus` pairs a `PlayerState` with the `PlaybackGeneration` it belongs to. The stream opens with the current pair, then behaves like `stateTransitions`.

`PlaybackGeneration` is opaque identity — `Hashable`, `Comparable`, no arithmetic. It is introduced as a named type rather than a raw `UInt64` because #81, #85 and #96 all need the same vocabulary, and `UInt64` could not later become a struct without breaking source.

## Published from both funnels

A state change keeps the same generation; a media change keeps the same state. Publishing from either alone would leave the stream describing a pair that was never true. So the status is republished from `publishPlaybackState` **and** from all four generation advance sites: `load(_:)`, the replacement branch of `play(_:)`, `recast(to:)`, and the `.mediaChanged` handler.

## `stateTransitions` is untouched

It still does not replay, and a test pins that. `recast(to:)` captures it **before** calling `play()` and awaits the next `.playing` — a replayed one from the outgoing session would let it report success before the replacement session ever started. That is why replay is a separate stream rather than a change to the existing one.

## Validation

- Six tests: late-subscriber replay, generation advance on load, same state across two sessions carrying different generations, `stateTransitions` non-replay, generation ordering, and stream termination.
- Negative control: swapping the replay subscription for a plain one fails all six.
- `swift test --no-parallel`: **1782 tests pass** against the locally built engine.
- swiftformat, swiftlint, doc coverage (100%, 730 symbols) and `-strict-concurrency=complete` all clean locally.

`Player.swift` crossed the 1000-line lint limit, so the new surface lives in `Player+PlaybackStatus.swift`.

## Scope

#89 is not closed here. Criterion 1 (buffering and error exactly once) and criterion 4 (`recast` observing async failure promptly) were already satisfied by earlier work; criterion 3 — starting a new generation without relying on a repeated raw `.playing` — is now *possible* for consumers but the internal `recast` path still keys off `.playing`, which is follow-up work.

### CODE REVIEW

File: Sources/SwiftVLC/Player/Player.swift [hunk 1]
```diff
@@ -417,6 +417,7 @@ public final class Player {
   nonisolated let playbackIntentBridge: Broadcaster<Bool>
   /// Carries normalized lifecycle transitions. See ``stateTransitions``.
   nonisolated let stateTransitionBridge = Broadcaster<PlayerState>(defaultBufferSize: 64)
+  nonisolated let playbackStatusBridge = Broadcaster<PlaybackStatus>(defaultBufferSize: 64)
```
[the remark this answers is not in the document]
Comment (omaralbeik) [reply Sources/SwiftVLC/Player/Player.swift:420]: Correct, and a real gap rather than just a doc mismatch. Fixed by seeding in `init`.

Every test in the suite loaded media first, so all of them missed a player that has never published anything. Worth stating: the failure mode was an indefinite wait, not a wrong value, since a subscriber to an idle player would sit on a stream documented to answer immediately.

While fixing it I also found the tests themselves were badly shaped for this. They awaited an iterator, so a missing replay hung the test rather than failing it: my first negative control for this fix ran until I killed it and reported nothing. They now terminate the bridge and drain, and the same control fails in 0.055s.
